### PR TITLE
Add STM32F769I-Discovery and enable double precision float support

### PR DESCRIPTION
**Pull request by strongly-typed**
Probably best to merge after #168, because it is based on this PR. Or I could rebase once #168 is merged.

### CODE REVIEW

File: src/xpcc/architecture/platform/board/stm32f769i_discovery/board.cfg [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[build]
+device = stm32f746ngh6u
```
Comment (salkinium) [inline src/xpcc/architecture/platform/board/stm32f769i_discovery/board.cfg]: ?

Comment (salkinium) [inline src/xpcc/architecture/platform/board/stm32f769i_discovery/board.cfg]: Ah, changed in the next commit. Never mind.

File: src/xpcc/architecture/platform/devices/stm32/stm32f765_767_768_769_777_778_779-a_b_i_n_v_z-g_i.xml [hunk 2]
```diff
@@ -8,7 +8,7 @@
     <flash device-size-id="g">1048576</flash>
     <flash device-size-id="i">2097152</flash>
     <ram>610304</ram>
-    <core>cortex-m7f</core>
+    <core>cortex-m7fd</core>
```
Comment (salkinium) [inline src/xpcc/architecture/platform/devices/stm32/stm32f765_767_768_769_777_778_779-a_b_i_n_v_z-g_i.xml:11]: Note to self: This should be somehow generated by the DFG.

**Comment by salkinium** [comment]
I'm going to merge #168 only with the commits up to and incl. ce774c6.
Everything following the addition of the device file will be merged here.

Could you please reorder the commits as follows to reduce their interdependency?
1. Toolchain and source changes related to cortex-m7fd (partly last commit)
2. Add device file with cortex-m7fd core tag
3. CAN id additions
4. Release tests
5. Discovery Board (keep board addition and example separate)
6. Example for board

**Comment by salkinium** [comment]
Note the new f7 device file is available from my [`feature/new_f7_device_file` branch](https://github.com/salkinium/xpcc/tree/feature/new_f7_device_file).

File: src/xpcc/architecture/platform/devices/stm32/stm32f765_767_768_769_777_778_779-a_b_i_n_v_z-g_i.xml (hunk 2 above)
Comment (salkinium) [reply src/xpcc/architecture/platform/devices/stm32/stm32f765_767_768_769_777_778_779-a_b_i_n_v_z-g_i.xml:11]: This is now generated by the DFG in #168.

**Comment by strongly-typed** [comment]
Alright! Are you OK with these commits?

**Comment by salkinium** [comment]
Awesome, much better!

**Comment by salkinium** [comment]
Merged manually. (I added the new F7 Discovery board to the Readme).

**Pull request closed without merging**